src/cr/cube/matrix/assembler.py

```diff
@@ -413,9 +425,23 @@ def rows_margin_proportion(self):
             )
 
         # --- otherwise rows-margin is a vector ---
-        return self._assemble_vector(
+        vector_components = SummedMarginalSubtotal(
             self._cube_result_matrix.rows_margin / denominator,
             self._row_subtotals,
+        )
+        return self._assemble_vector(
+            vector_components.blocks,
+            self._row_order,
+        )
+
+    @lazyproperty
+    def rows_scale_median(self):
+        """Optional 1D Row scale median margin for this cube-result"""
+        if not self._measures.scaled_counts.column_has_numeric_values:
+            return None
+
+        return self._assemble_vector(
+            self._measures.scaled_counts.rows_scale_median,
             self._row_order,
         )
```

Comment:
Okay, I think maybe you're right about where to do this test, that is here in `Assembler` rather than further down. So the interaction model would be like this:

```python
>>> scaled_counts = self._measures.scaled_counts
>>> scaled_counts.columns_have_numeric_values
False
>>> scaled_counts.rows_scale_median
ValueError("`.rows_scale_median` is undefined on a matrix with no column element numeric values")
```

Btw, I think it should be `.columns_have_...` instead of `.column_has_...`.

So here in assembler we check before accessing and thereby avoid an exception. In sort by value we catch the exception and return payload order instead. That seems right to me.

Reply:
One other thing on this, the return value needs to say something like `.rows_scale_median_blocks` rather than just `.rows_scale_median`. The latter sounds like either an object or an assembled vector or whatever; I think for readability's sake we need to make it explicit this is a collection (pair in this case). "blocks" comes up a lot elsewhere so should be a familiar term to a reader.